plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -96,18 +96,24 @@ func (c *X509Cert) getCert(u *url.URL, timeout time.Duration) ([]*x509.Certifica
 		if err != nil {
 			return nil, err
 		}
+		var certs []*x509.Certificate
+		for {
+			block, rest := pem.Decode(content)
+			if block == nil {
+				return nil, fmt.Errorf("failed to parse certificate PEM")
```

Comment:
I'm concerned this could get hit if there is trailing whitespace after the final block.  Can you add a test for this?

Reply:
Added trim for whitespaces because it resulted in error and added some test cases